### PR TITLE
Add vpc constraint

### PR DESCRIPTION
**Pull request by tthyer**
Constrains the linux product to one vpc
Constrains the linux product subnet to subnets of that vpc

### CODE REVIEW

File: config/prod/sc-vpc-constraint.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+Description: Template Constraint for VPC.
```
Comment (zaro0508) [inline config/prod/sc-vpc-constraint.yaml]: shouldn't this file be in `templates` folder?

File: config/prod/sc-vpc-constraint.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+AWSTemplateFormatVersion: 2010-09-09
+Resources:
+  TemplateConstraint:
+    Type: AWS::ServiceCatalog::LaunchTemplateConstraint
+    Properties:
+      AcceptLanguage: en
```
Comment (zaro0508) [inline config/prod/sc-vpc-constraint.yaml]: maybe we should pass these in as parameters?

File: config/prod/sc-vpc-constraint.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      Description: Limit VPC selection.
+      PortfolioId: port-z2m47jv6sckgi
+      ProductId: prod-zrb4keqx67iwk
+      Rules: >-
```
Comment (zaro0508) [inline config/prod/sc-vpc-constraint.yaml]: if you think these rules can be re-used in multiple CFN templates then maybe you'll want to include them from external files? https://stackoverflow.com/questions/52277610/can-i-insert-text-from-another-file-into-my-cloudformation-template

File: config/prod/sc-vpc-constraint.yaml (hunk 1 above)
Comment (tthyer) [reply config/prod/sc-vpc-constraint.yaml]: 🤦‍♀ yes

File: config/prod/sc-vpc-constraint.yaml (hunk 2 above)
Comment (tthyer) [reply config/prod/sc-vpc-constraint.yaml]: yes

File: config/prod/sc-vpc-constraint.yaml (hunk 3 above)
Comment (tthyer) [reply config/prod/sc-vpc-constraint.yaml]: maybe, I'll think about it

File: config/prod/sc-vpc-constraint.yaml (hunk 2 above)
Comment (tthyer) [reply config/prod/sc-vpc-constraint.yaml]: @zaro0508 passing in product id and portfolio id as parameters works fine, but I cannot use any parameters that I set in the constraint template itself into its rules

File: config/prod/sc-vpc-constraint.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+template_path: "sc-vpc-constraint.yaml"
+stack_name: "sc-vpc-constraint-on-portfolio-ec2-product-linux"
+parameters:
+  PortfolioID: port-z2m47jv6sckgi
```
Comment (zaro0508) [inline config/prod/sc-vpc-constraint.yaml]: you mind adding a comment identifying which portfolio and product these are?

File: templates/sc-vpc-constraint.yaml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        {
+          "LimitVPC": {
+            "Assertions": [
+              {
+                "Assert": {
+                  "Fn::Equals": [
+                    {
+                      "Ref": "VPC"
+                    },
+                    "vpc-0d92dc681a4078458"
```
Comment (zaro0508) [inline templates/sc-vpc-constraint.yaml]: would it make sense to pass this in as well?

**Pull request merged**